src/components/Form.js

```diff
@@ -1,12 +1,36 @@
-const Form = () => (
-  <div className="form">
-    <form>
-      <h3>Add a Book</h3>
-      <input type="text" name="title" placeholder="title" />
-      <input type="text" name="author" placeholder="author" />
-      <button type="submit">Add Book</button>
-    </form>
-  </div>
-);
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/bookslice';
+
+const Form = () => {
+  const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+
+  function addBookHandler(event) {
+    event.preventDefault();
+
+    const newBook = {
+      id: Math.floor(Math.random() * 1000),
+      title,
+      author,
+    };
+
+    dispatch(addBook(newBook));
+
+    setTitle('');
+    setAuthor('');
+  }
+  return (
+    <div className="form">
+      <form>
+        <h3>Add a Book</h3>
+        <input type="text" name="title" placeholder="title" />
```

Comment:
- [ ] Your inputs are missing an onChange handler that will update the state of title to be dispatched with addBook. This is causing your addBook functionality to not work.